clang/lib/CIR/CodeGen/CIRGenFunction.cpp

```diff
@@ -213,6 +222,11 @@ mlir::LogicalResult CIRGenFunction::emitFunctionBody(const clang::Stmt *body) {
 cir::FuncOp CIRGenFunction::generateCode(clang::GlobalDecl gd, cir::FuncOp fn,
                                          cir::FuncType funcType) {
   const auto funcDecl = cast<FunctionDecl>(gd.getDecl());
+  fnRetQualTy = funcDecl->getReturnType();
```

Comment:
This whole thing doesn't seem useful as far as I can tell, we should just be able to use the return type in `startFunction`